Add integration test for Ceph RGW multisite sync policies feature

ionutbalutoiu opened the pull request:
Add Ceph RGW multisite sync policies integration test.

Add integration test for the Multisite sync policies feature proposed to the
upstream `ceph-radosgw` Juju charm:

* https://review.opendev.org/c/openstack/charm-specs/+/899579
* https://review.opendev.org/c/openstack/charm-ceph-radosgw/+/899581

ionutbalutoiu commented:
/cc @UtkarshBhatthere

UtkarshBhatthere commented on zaza/openstack/charm_tests/ceph/tests.py:
```diff
@@ -1038,7 +1038,178 @@ def test_003_object_storage_and_secondary_block(self):
                                               'active')
 
-    def test_004_migration_and_multisite_failover(self):
+    def test_004_multisite_directional_sync_policy(self):
```
The new testcase added in [This PR](https://github.com/openstack-charmers/zaza-openstack-tests/pull/1176/) is also numbered `004`. The numbering here is used to run the tests in a particular order and It would be preferable to have non-conflicting test numbers.

UtkarshBhatthere reviewed (approved):
LGTM

ionutbalutoiu commented:
> LGTM

I rebased this PR against latest upstream main branch.

I noticed that a new test case (`test_005_virtual_hosted_bucket`) was added after `test_004_migration_and_multisite_failover`.

Besides renaming `test_004_migration_and_multisite_failover` to `test_100_migration_and_multisite_failover` as discussed previously (to have the multi-site scaledown scenario run last), I also renamed `test_005_virtual_hosted_bucket` to `test_101_virtual_hosted_bucket` to keep the same test order.

See commit [CephRGWTest: Reorder test cases](https://github.com/openstack-charmers/zaza-openstack-tests/pull/1193/commits/c0f04747620bfd2d62dd50959c49a96919b90b84) from this PR.

Besides the tests re-order, I also updated the new test case `test_004_multisite_directional_sync_policy` with my findings regarding the CI failures with `focal-yoga-multisite` (which installs Ceph Quincy release). More details in [this comment](https://review.opendev.org/c/openstack/charm-ceph-radosgw/+/899581/comments/334709d1_fb952e59) from `charm-ceph-radosgw` patchset.

ionutbalutoiu commented:
> > LGTM
> 
> I rebased this PR against latest upstream main branch.
> 
> I noticed that a new test case (`test_005_virtual_hosted_bucket`) was added after `test_004_migration_and_multisite_failover`.
> 
> Besides renaming `test_004_migration_and_multisite_failover` to `test_100_migration_and_multisite_failover` as discussed previously (to have the multi-site scaledown scenario run last), I also renamed `test_005_virtual_hosted_bucket` to `test_101_virtual_hosted_bucket` to keep the same test order.
> 
> See commit [CephRGWTest: Reorder test cases](https://github.com/openstack-charmers/zaza-openstack-tests/pull/1193/commits/c0f04747620bfd2d62dd50959c49a96919b90b84) from this PR.
> 
> Besides the tests re-order, I also updated the new test case `test_004_multisite_directional_sync_policy` with my findings regarding the CI failures with `focal-yoga-multisite` (which installs Ceph Quincy release). More details in [this comment](https://review.opendev.org/c/openstack/charm-ceph-radosgw/+/899581/comments/334709d1_fb952e59) from `charm-ceph-radosgw` patchset.

Also, I included a [small fix](https://github.com/openstack-charmers/zaza-openstack-tests/pull/1193/commits/e68c902fc868f6631b2c447c2a493b98a1e13887) that might cause failures sometimes (it happened on my local Zaza env).

UtkarshBhatthere reviewed (approved):
LGTM, Thanks for the tests @ionutbalutoiu

sabaini commented:
Thanks, lgtm! -- Merged
